### DISCUSSION TITLE
My custom dataset has mother_ Tiger and push_ Tiger has two categories, mother during reasoning_ Tiger did not successfully annotate.

### DISCUSSION THREAD

**Discussion by SCsleepy** [open-mmlab/mmpose#2566] [Q&A]
The model I used is dekr+hrnet, with a custom dataset training set of 90 images and a validation set of 10 images. Each image has four little tigers and one big tiger. I trained the weight inference of 500epoch (ap=0.31) and found that only the small tiger was labeled and the big tiger was not labeled. I trained the annotation image and inference result image as follows. I want to know what caused it.
![F2`MEIZ5{(}D9TNW 8MCT28](https://github.com/open-mmlab/mmpose/assets/117082999/295e05f5-a1ea-4a2e-8834-3d1b599fcf0c)
![2G586W{6KCV9@{(GH{FB2I](https://github.com/open-mmlab/mmpose/assets/117082999/85ca0daa-e473-473c-9c16-8c35608fc4b0)

**Comment by SCsleepy** [comment]
The dataset is in Coco format, and the following is an excerpt from the dataset
"images": [
  {
   "file_name": "1012_002251.jpg",
   "height": 1440,
   "width": 2560,
   "id": 5
  },
.....
"annotations": [
  {
   "segmentation": [
    [
     533.37,
     131.01,
     533.06,
     128.98,
     533.21,
     127.73,
     532.12,
     126.33,
     529.31,
     126.01,
     528.21,
     127.57,
     528.21,
     129.29,
     528.99,
     130.23,
     528.99,
     130.7,
     528.37,
     132.1,
     526.18,
     134.29,
     524.31,
     136.79,
     522.9,
     139.29,
     521.81,
     142.41,
     521.81,
     144.6,
     522.59,
     145.22,
     522.9,
     145.69,
     525.4,
     143.66,
     527.28,
     142.41,
     529.15,
     140.69,
     530.87,
     138.35,
     532.43,
     137.1,
     533.37,
     135.85,
     533.84,
     133.67,
     533.84,
     132.1
    ],
    [
     524.78,
     147.88,
     524.31,
     151.16,
     524.31,
     154.28,
     524.15,
     158.03,
     524.47,
     161,
     524.93,
     164.28,
     525.4,
     168.34,
     525.25,
     170.83,
     524.47,
     172.86,
     522.75,
     173.8,
     522.59,
     174.43,
     529.15,
     174.74,
     529.46,
     173.8,
     531.18,
     172.4,
     531.81,
     171.62,
     532.12,
     168.34,
     531.18,
     155.84,
     531.34,
     149.6,
     531.96,
     146,
     531.49,
     145.85,
     527.59,
     148.35,
     526.5,
     147.88,
     525.25,
     147.25
    ]
   ],
   "keypoints": [
    1425.302987,
    141.4731466,
    2,
    0,
    0,
    0,
    1401.679555,
    169.1721407,
    2,
    1316.746787,
    135.1248356,
    2,
    1276.117565,
    183.3720357,
    2,
    0,
    0,
    0,
    1378.960281,
    213.20912,
    2,
    1207.555755,
    137.664162,
    2,
    1307.859145,
    216.3832779,
    2,
    0,
    0,
    0,
    0,
    0,
    0,
    1316.746787,
    87.51246713,
    2,
    1246.915313,
    65.29336182,
    2,
    1168.831028,
    103.3832566,
    2,
    1152.325407,
    134.4900041,
    2,
    1241.201829,
    203.0518147,
    2
   ],
   "num_keypoints": 12,
   "area": 41244.15965392148,
   "iscrowd": 0,
   "image_id": 5,
   "bbox": [
    1152.325407,
    65.29336182,
    1425.302987,
    216.3832779
   ],
   "category_id": 2,
   "id": 26
  },
  {
   "segmentation": [
    [
     533.37,
     131.01,
     533.06,
     128.98,
     533.21,
     127.73,
     532.12,
     126.33,
     529.31,
     126.01,
     528.21,
     127.57,
     528.21,
     129.29,
     528.99,
     130.23,
     528.99,
     130.7,
     528.37,
     132.1,
     526.18,
     134.29,
     524.31,
     136.79,
     522.9,
     139.29,
     521.81,
     142.41,
     521.81,
     144.6,
     522.59,
     145.22,
     522.9,
     145.69,
     525.4,
     143.66,
     527.28,
     142.41,
     529.15,
     140.69,
     530.87,
     138.35,
     532.43,
     137.1,
     533.37,
     135.85,
     533.84,
     133.67,
     533.84,
     132.1
    ],
    [
     524.78,
     147.88,
     524.31,
     151.16,
     524.31,
     154.28,
     524.15,
     158.03,
     524.47,
     161,
     524.93,
     164.28,
     525.4,
     168.34,
     525.25,
     170.83,
     524.47,
     172.86,
     522.75,
     173.8,
     522.59,
     174.43,
     529.15,
     174.74,
     529.46,
     173.8,
     531.18,
     172.4,
     531.81,
     171.62,
     532.12,
     168.34,
     531.18,
     155.84,
     531.34,
     149.6,
     531.96,
     146,
     531.49,
     145.85,
     527.59,
     148.35,
     526.5,
     147.88,
     525.25,
     147.25
    ]
   ],
   "keypoints": [
    1811.28051,
    220.827099,
    2,
    1863.971531,
    312.8776781,
    2,
    1787.15691,
    295.1023938,
    2,
    0,
    0,
    0,
    0,
    0,
    0,
    1895.71311,
    262.7259832,
    2,
    1879.842321,
    325.5743097,
    2,
    1800.488373,
    140.8383199,
    2,
    1732.561394,
    189.0855199,
    2,
    0,
    0,
    0,
    1771.920952,
    185.2765305,
    2,
    1859.52771,
    217.0181095,
    2,
    1863.971531,
    178.2933831,
    2,
    1862.067036,
    124.3326988,
    2,
    0,
    0,
    0,
    1806.836689,
    111.0012356,
    2
   ],
   "num_keypoints": 12,
   "area": 35007.96524681013,
   "iscrowd": 0,
   "image_id": 5,
   "bbox": [
    1732.561394,
    111.0012356,
    1895.71311,
    325.5743097
   ],
   "category_id": 2,
   "id": 27
  },
  {
   "segmentation": [
    [
     533.37,
     131.01,
     533.06,
     128.98,
     533.21,
     127.73,
     532.12,
     126.33,
     529.31,
     126.01,
     528.21,
     127.57,
     528.21,
     129.29,
     528.99,
     130.23,
     528.99,
     130.7,
     528.37,
     132.1,
     526.18,
     134.29,
     524.31,
     136.79,
     522.9,
     139.29,
     521.81,
     142.41,
     521.81,
     144.6,
     522.59,
     145.22,
     522.9,
     145.69,
     525.4,
     143.66,
     527.28,
     142.41,
     529.15,
     140.69,
     530.87,
     138.35,
     532.43,
     137.1,
     533.37,
     135.85,
     533.84,
     133.67,
     533.84,
     132.1
    ],
    [
     524.78,
     147.88,
     524.31,
     151.16,
     524.31,
     154.28,
     524.15,
     158.03,
     524.47,
     161,
     524.93,
     164.28,
     525.4,
     168.34,
     525.25,
     170.83,
     524.47,
     172.86,
     522.75,
     173.8,
     522.59,
     174.43,
     529.15,
     174.74,
     529.46,
     173.8,
     531.18,
     172.4,
     531.81,
     171.62,
     532.12,
     168.34,
     531.18,
     155.84,
     531.34,
     149.6,
     531.96,
     146,
     531.49,
     145.85,
     527.59,
     148.35,
     526.5,
     147.88,
     525.25,
     147.25
    ]
   ],
   "keypoints": [
    1717.104888,
    1040.819085,
    2,
    1623.149814,
    1046.532569,
    2,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    1608.548688,
    1153.184274,
    2,
    1576.172277,
    1098.588758,
    2,
    0,
    0,
    0,
    0,
    0,
    0,
    1609.183519,
    1365.218022,
    2,
    1587.599246,
    1356.965212,
    2,
    1663.144204,
    1139.852811,
    2,
    1653.62173,
    1242.060696,
    2,
    1658.065551,
    1364.583191,
    2,
    1649.177909,
    1407.751738,
    2,
    0,
    0,
    0
   ],
   "num_keypoints": 10,
   "area": 51712.77684844703,
   "iscrowd": 0,
   "image_id": 5,
   "bbox": [
    1576.172277,
    1040.819085,
    1717.104888,
    1407.751738
   ],
   "category_id": 2,
   "id": 28
  },
  {
   "segmentation": [
    [
     533.37,
     131.01,
     533.06,
     128.98,
     533.21,
     127.73,
     532.12,
     126.33,
     529.31,
     126.01,
     528.21,
     127.57,
     528.21,
     129.29,
     528.99,
     130.23,
     528.99,
     130.7,
     528.37,
     132.1,
     526.18,
     134.29,
     524.31,
     136.79,
     522.9,
     139.29,
     521.81,
     142.41,
     521.81,
     144.6,
     522.59,
     145.22,
     522.9,
     145.69,
     525.4,
     143.66,
     527.28,
     142.41,
     529.15,
     140.69,
     530.87,
     138.35,
     532.43,
     137.1,
     533.37,
     135.85,
     533.84,
     133.67,
     533.84,
     132.1
    ],
    [
     524.78,
     147.88,
     524.31,
     151.16,
     524.31,
     154.28,
     524.15,
     158.03,
     524.47,
     161,
     524.93,
     164.28,
     525.4,
     168.34,
     525.25,
     170.83,
     524.47,
     172.86,
     522.75,
     173.8,
     522.59,
     174.43,
     529.15,
     174.74,
     529.46,
     173.8,
     531.18,
     172.4,
     531.81,
     171.62,
     532.12,
     168.34,
     531.18,
     155.84,
     531.34,
     149.6,
     531.96,
     146,
     531.49,
     145.85,
     527.59,
     148.35,
     526.5,
     147.88,
     525.25,
     147.25
    ]
   ],
   "keypoints": [
    2067.290874,
    1336.436316,
    2,
    2184.890178,
    1339.844945,
    2,
    2076.756227,
    1437.407016,
    2,
    1937.201772,
    1272.619467,
    2,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    1780.31929,
    889.190065,
    2,
    1522.229628,
    1011.693453,
    2,
    1780.31929,
    987.9063873,
    2,
    0,
    0,
    0,
    2102.634029,
    1210.315451,
    2,
    2061.006664,
    999.7999201,
    2,
    1970.615815,
    822.5862813,
    2,
    1854.059193,
    746.4676713,
    2,
    1720.851626,
    526.4373144,
    2
   ],
   "num_keypoints": 12,
   "area": 603663.683495592,
   "iscrowd": 0,
   "image_id": 5,
   "bbox": [
    1522.229628,
    526.4373144,
    2184.890178,
    1437.407016
   ],
   "category_id": 1,
   "id": 30
  },
.... 
"categories": [
  {
   "id": 1,
   "name": "mother_tiger"
  },
  {
   "id": 2,
   "name": "pup_tiger"
  }

**Comment by Ben-Louis** [comment]
Hi, thanks for using MMPose. It seems like there may be a misdetection issue. There are a couple of possible reasons for this:
1. The [rescore net](the https://github.com/open-mmlab/mmpose/blob/537bd8e543ab463fb55120d5caaa1ae22d6aaf06/configs/body_2d_keypoint/dekr/coco/dekr_hrnet-w32_8xb10-140e_coco-512x512.py#L101C6-L101C6) from DEKR may not be suitable for your dataset, and should be removed.
2. If your training set is too small, it may not be able to effectively train a large network like HRNet. Consider training DEKR on a larger dataset like AP10k, and then fine-tuning it with your own data.
